skglm/solvers/gram_cd.py

```diff
@@ -132,6 +137,12 @@ def solve(self, X, y, datafit, penalty, w_init=None, Xw_init=None):
             p_objs_out.append(p_obj)
         return w, np.array(p_objs_out), stop_crit
 
+    def custom_compatibility_check(self, X, y, datafit, penalty):
+        if not isinstance(datafit, Quadratic):
+            raise AttributeError(
+                f"`GramCD` supports only `Quadratic` datafit, got {datafit}."
```

Comment:
```suggestion
                f"`GramCD` supports only `Quadratic` datafit, got {datafit.__class__.__name__}."
```
otherrwise error message is: 
`AttributeError: `GramCD` supports only `Quadratic` datafit, got <numba.experimental.jitclass.boxing.Poisson object at 0x7f4bb1fa3f40>`